ssz_serialization/merkleization.nim

```diff
@@ -635,6 +721,248 @@ func hashTreeRootAux[T](x: T): Digest =
   else:
     unsupported T
 
+func hashTreeRootAux[T](
```

Comment:
It would be useful to provide some references to the spec being implemented here.

Reply:
It is the same spec as the rest of the file is using, i.e., https://github.com/ethereum/consensus-specs/tree/dev/ssz

The only difference is that the overload allows querying for `GeneralizedIndex != 1`, as in getting intermediate hashes out.

Reply:
Spec link is actually already there, at the top of the file.